### PR TITLE
Bratseth/propagate cluster info

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

**Review by mpolden — COMMENTED** [review]
You probably know this, but the implementations must be updated in the internal repo together with this.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/configserver/Application.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * @author bratseth
+ */
+public class Application {
+
+    private ApplicationId id;
+    private Map<ClusterSpec.Id, Cluster> clusters;
+
+    public Application(ApplicationId id, Collection<Cluster> clusters) {
+        this.id = id;
+        this.clusters = clusters.stream().collect(Collectors.toMap(c -> c.id(), c -> c));
```
[the remark this answers is not in the document]
Comment (bratseth) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/configserver/Application.java]: Done, thanks

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/configserver/Cluster.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * @author bratseth
+ */
+public class Cluster {
+
+    private ClusterSpec.Id id;
+    private ClusterResources min;
+    private ClusterResources max;
+    private ClusterResources current;
+    private Optional<ClusterResources> target;
+    private Optional<ClusterResources> suggested;
```
[the remark this answers is not in the document]
Comment (bratseth) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/configserver/Cluster.java]: Ok, done

**Review by mpolden — APPROVED** [review]

**Comment by bratseth** [comment]
> You probably know this, but the implementations must be updated in the internal repo together with this.

Yes, you have that PR too.

How do you feel about doing all the wire classes like this?

**Pull request merged**